src/main/kotlin/fi/metatavu/timebank/api/controllers/SynchronizeController.kt

```diff
@@ -35,35 +35,65 @@ class SynchronizeController {
     private val oldestSyncDate = LocalDate.parse("2021-07-30")
 
     /**
-     * Synchronizes time entries from Forecast to Time-bank
+     * Begins synchronization
      *
      * @param after YYYY-MM-DD LocalDate
+     * @param syncEntries run synchronization fun
+     * @param syncDeletedEntries run synchronizeDeletions fun
      */
-    suspend fun synchronize(after: LocalDate? = oldestSyncDate) {
-        try {
-            var forecastPersons = personsController.getPersonsFromForecast()
+    suspend fun startSynchronization(after: LocalDate? = oldestSyncDate, syncEntries: Boolean = false, syncDeletedEntries: Boolean = false) {
+        var forecastPersons = personsController.getPersonsFromForecast()
 
-            forecastPersons.forEach { worktimeCalendarController.checkWorktimeCalendar(it) }
+        forecastPersons.forEach { worktimeCalendarController.checkWorktimeCalendar(it) }
 
-            forecastPersons = personsController.filterPersons(forecastPersons)
+        forecastPersons = personsController.filterPersons(forecastPersons)
 
-            val tasks = retrieveAllTasks()
-            val entries = retrieveAllEntries(after = after, forecastPersons = forecastPersons)
+        val entries = retrieveAllEntries(
+            after = after,
+            forecastPersons = forecastPersons
+        )
+
+        if (syncEntries) {
+            synchronize(
+                after = after,
+                forecastPersons = forecastPersons,
+                forecastTimeEntries = entries
+            )
+        }
+        if (syncDeletedEntries) {
+            synchronizeDeletions(
+                after = after,
+                forecastPersons = forecastPersons,
+                forecastTimeEntries = entries
+            )
+        }
+    }
+
+    /**
+     * Synchronizes time entries from Forecast to Time-bank
+     *
+     * @param after YYYY-MM-DD LocalDate
+     * @param forecastPersons list of ForecastPersons
+     * @param forecastTimeEntries list of ForecastTimeEntries
+     */
+    suspend fun synchronize(after: LocalDate? = oldestSyncDate, forecastPersons: List<ForecastPerson>, forecastTimeEntries: List<ForecastTimeEntry>) {
```

Comment:
This methods after parameter doesn't need the default value anymore due to the new `startSynchronization` method already defaulting it to `oldestSyncDate`